core/domain/skill_services.py

```diff
@@ -390,76 +395,72 @@ def filter_skills_by_status(augmented_skill_summaries, status):
     elif status == constants.SKILL_STATUS_OPTIONS['ASSIGNED']:
         assigned_augmented_skill_summaries = []
         for augmented_skill_summary in augmented_skill_summaries:
-            if augmented_skill_summary.topic_name:
+            if augmented_skill_summary.topic_name is not None:
                 assigned_augmented_skill_summaries.append(
                     augmented_skill_summary)
-
         return assigned_augmented_skill_summaries
 
 
-def filter_skills_by_classroom(augmented_skill_summaries, classroom_name):
+def _filter_skills_by_classroom(augmented_skill_summaries, classroom_name):
     """Returns the skill summary dicts after filtering by classroom_name.
 
     Args:
         augmented_skill_summaries: list(AugmentedSkillSummary).
             The list of augmented skill summaries.
-        classroom_name: string. The name of the classroom.
+        classroom_name: str. The classroom_name of the topic to which skill is
+            assigned to.
 
     Returns:
         list(AugmentedSkillSummary). The list of augmented
             skill summaries with the given classroom name.
     """
 
-    if not classroom_name or classroom_name == 'All':
+    if classroom_name is None or classroom_name == 'All':
         return augmented_skill_summaries
 
     augmented_skill_summaries_with_classroom_name = []
     for augmented_skill_summary in augmented_skill_summaries:
-        if (augmented_skill_summary.classroom_name and
-                augmented_skill_summary.classroom_name == classroom_name):
+        if augmented_skill_summary.classroom_name == classroom_name:
             augmented_skill_summaries_with_classroom_name.append(
                 augmented_skill_summary)
 
     return augmented_skill_summaries_with_classroom_name
 
 
-def is_keyword_present_in_skill(augmented_skill_summary, keywords):
+def _is_keyword_present_in_skill(augmented_skill_summary, keywords):
     """Returns whether the keywords match the skill description.
 
     Args:
-        augmented_skill_summary: list(AugmentedSkillSummary).
-            The augmented skill summaries.
+        augmented_skill_summary: list(AugmentedSkillSummary). The augmented
+            skill summaries.
         keywords: list(str). The keywords to match.
 
     Returns:
         bool. A boolean indicating if any of the keywords exist in the given
-        augmented skill summary.
+            augmented skill summary.
     """
-    for keyword in keywords:
-        if augmented_skill_summary.description.lower().find(
-                keyword.lower()) != -1:
-            return True
-    return False
+    return any((augmented_skill_summary.description.lower().find(
```

Comment:
In a previous comment, I suggested

> Also I think you can probably inline this? (once it becomes a one-line command)

What this ("inlining") means is, get rid of this function and move the logic to the caller. You don't need to make separate functions for short things, unless they're reused elsewhere.

Reply:
Oh, understood. Done.